### PR TITLE
Pass dbt vars to dbt

### LINKED ISSUE #1262: Allow for DBT arguments in config
The current DBT config assumes no arguments are passed on ```dbt run```, but many instances where the dbt-profile is configured to uses variables when building, such as for the database name. This prevents  sqlfluff from  being able to build the targets.

### PR DESCRIPTION
**Pull request by tcholewik**
Two small changes related to dbt plugin, both have been documented and tested, but I would be happy to take another look if something does not look right.

## Fixed issue #1262
fixes #1262

Allows sqlfluff to emulate dbt's cli calls with vars flag:
```
dbt run --vars 'variable_not_included_in_dbt_config: 1'
```

If dbt project does not have all of its vars(without defaults) in `dbt_project.yml`, dbt throws the following error:
```
Could not render {{ var('variable_not_included_in_dbt_config') }}: Required var 'variable_not_included_in_dbt_config' not found in config:
  Vars supplied to <Configuration> = {}
```

The way users can include those variables is through `.sqlfuff` in `[templates:dbt:contex]` section( similar to what we do for jinja template).

## Docker Dev container Fixes

 * `CONTRIBUTING.md` mentions that `dbt_profiles.yml` need to be manually updated to work with dev container until a fix comes - this fix just arrived
 *  current Dockerfile uses options that require enabling Buildkit first so I added them to `startup` script
 * `docker-compose` file uses `platform` so I bumped the file to the version that introduced this setting

### CODE REVIEW

**Comment by barrywhart** [comment]
@tcholewik: Lots of tests are failing. Can you take a look?

**Comment by tcholewik** [comment]
Hi @barrywhart  thanks for taking a look at my PR, I've been looking into this for the past couple of days I'll try to push the fix shortly.

**Comment by tcholewik** [comment]
@barrywhart I made changes and pushed them. I was able to run tox + git workflow locally without errors.

I was under impression that CI workflows kick off at 2 am every night but nothing ran here last night?

**Comment by tunetheweb** [comment]
> I was under impression that CI workflows kick off at 2 am every night but nothing ran here last night?

It kicks off on every commit. But for new contributors a maintainer has to approve the run. I've approved it there now and it's running.

File: plugins/sqlfluff-templater-dbt/test/templater_test.py
Comment (tunetheweb) [inline plugins/sqlfluff-templater-dbt/test/templater_test.py:442]: Keep black happy:

```suggestion
@pytest.mark.parametrize(
    ("model_path", "var_value"),
    [
        ("models/vars_from_cli.sql", "expected_value"),
        ("models/vars_from_cli.sql", [1]),
        ("models/vars_from_cli.sql", {"nested": 1}),
    ],
)
```

File: plugins/sqlfluff-templater-dbt/test/templater_test.py
Comment (tunetheweb) [inline plugins/sqlfluff-templater-dbt/test/templater_test.py:445]: More black:

```suggestion
        project_dir, dbt_templater, model_path, var_value
):  # noqa: F811
```

File: docs/source/configuration.rst
Comment (barrywhart) [inline docs/source/configuration.rst:637]: Typo: contex should be context, I think?

File: plugins/sqlfluff-templater-dbt/sqlfluff_templater_dbt/templater.py
Comment (barrywhart) [inline plugins/sqlfluff-templater-dbt/sqlfluff_templater_dbt/templater.py:235]: Nitpick: could we add a type annotation for the return type (str)?

**Review by barrywhart — CHANGES_REQUESTED** [review]
Looks pretty good overall! A few small suggestions. I'm reviewing on my phone. Would like to review and test locally on my notebook before final approval.

File: docs/source/configuration.rst
Comment (tcholewik) [reply docs/source/configuration.rst]: Sure I pushed a patch

File: plugins/sqlfluff-templater-dbt/sqlfluff_templater_dbt/templater.py
Comment (tcholewik) [reply plugins/sqlfluff-templater-dbt/sqlfluff_templater_dbt/templater.py]: Sure I pushed a patch

File: plugins/sqlfluff-templater-dbt/test/templater_test.py
Comment (barrywhart) [inline plugins/sqlfluff-templater-dbt/test/templater_test.py:451]: This looks like a possible bug in the test -- I think this code is actually making changes to `DBT_FLUFF_CONFIG`, which could break other tests. Please use `copy.deepcopy()` to copy the dictionary before making any changes to it.

**Review by barrywhart — CHANGES_REQUESTED** [review]

File: docs/source/configuration.rst
Comment (barrywhart) [inline docs/source/configuration.rst:638]:
```suggestion
See below configuration and its equivalent dbt command:
```

**Review by barrywhart — APPROVED** [review]

**Pull request merged**
